src/hardforks/hardforks.h

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2014-2019, The Monero Project
+//
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are
+// permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#pragma once
+
+#include <stdint.h>
+#include <time.h>
+
+struct hardfork_t
+{
+  uint8_t version;
+  uint64_t height;
+  uint8_t threshold;
+  time_t time;
```

Comment:
Can a `uint32_t` or `uint64_t` be used here instead? This type is platform dependent.

Reply:
It could, but it's also local only.

Reply:
The type and width are unspecified and there are hard-coded constants forced into a `time_t`. Luckily, the compiler is in C++11 mode which is required to fail if narrowing will occur, provided the structure is brace-initialized (which is the case currently). I suppose `time_t` helps provide indication to the expected values ... so I'm not stressing this much.

Although, after peeking at some of these other files, should this be in `src/cryptonote_basic/hardfork.{h|cpp}`?  There is an identical `struct` within that file with an identical purpose. The largest include in `hardfork.h`, `cryptonote_basic.h`, can be dropped with some forward declares. And the primary consumer of this information is the code in `hardfork.cpp`.

Reply:
The structs could be merged indeed, but this new file is the data that gets loaded into the existing hardfork class. The one in cryptonote_basic is lib code, so independent of the actual data. I guess it might not matter that much in practice...

Reply:
I feel the same as @vtnerd, it seems worthwhile to eliminate some redundant code. I think we can simply replace `HardFork::Params` with `hardfork_t` and `Blockchain::get_hard_fork_heights(MAINNET).back().version` in cryptonote_core.cpp with `(mainnet_hard_forks+num_mainnet_hard_forks-1)->version`, respectively, and remove the unneeded struct and function.